### PR TITLE
chore: Schematics - force feature configuration

### LINKED ISSUE #12289: Schematics - ensure the User feature are pre-configured when installing a dependent feature
As the title suggests, when installing a feature that depend on the User library, we need to make sure the User library is properly configured, and not just installed.

What this implies is that we should pass the User CLI feature flags when installing it as a dependency.
We should also print a message that we are "forced" to configure the User library, as it's a dependency of the feature that's being installed. Something along these lines: "INFO: configuring the User's [feature-name-here], as it's required for the [feature-that's-being-installed] to work properly.".

Will be used by: https://github.com/SAP/spartacus/issues/10817

### PR DESCRIPTION
**Pull request by znikola**
closes #12289

### CODE REVIEW

File: projects/schematics/src/shared/utils/lib-utils.ts [hunk 1]
```diff
@@ -603,29 +632,73 @@ export function addPackageJsonDependencies(
...
+      dependencyManagement.featureDependencies
+    );
+    addSchematicsTasks(featureOptions, context);
 
-  const featureOptions = createSpartacusFeatureOptionsForLibrary(
-    spartacusLibraries.map((dependency) => dependency.name),
-    options.options
-  );
-  addSchematicsTasks(featureOptions, options.context);
+    return installPackageJsonDependencies();
```
Comment (marlass) [inline projects/schematics/src/shared/utils/lib-utils.ts]: Do we need to install packages here? Everything should be installed after `addPackageJsonDependenciesForLibrary`

Comment (znikola) [reply projects/schematics/src/shared/utils/lib-utils.ts]: Removed

File: feature-libs/user/schematics/add-user/index.ts [hunk 2]
```diff
@@ -116,5 +112,11 @@ function addProfileFeature(options: SpartacusUserOptions): Rule {
       scssFileName: SCSS_FILE_NAME,
       importStyle: SPARTACUS_USER,
     },
+    dependencyManagement: {
+      featureName: CLI_USER_FEATURE,
```
Comment (marlass) [inline feature-libs/user/schematics/add-user/index.ts]: Should we use here the user profile feature? As the dependencies are for specific features, not libraries

Comment (znikola) [reply feature-libs/user/schematics/add-user/index.ts]: I agree. Done.

**Review by marlass — APPROVED** [review]

**Pull request merged**